### PR TITLE
Lecture9 homework

### PR DESCRIPTION
**Pull request by MariaOdinets**

### CODE REVIEW

**Comment by MariaOdinets** [comment]
HW 9
0.1, 0.2
1 (Without ProducrSet realization)

File: ConsoleApp1/HomeWork_lecture9_Collections/Product.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace TMS_homeworks.HomeWork_lecture9_Collections
  //Создать базу (List) из n товаров, вывести полную информацию из базы на экран, а также организовать поиск просроченного товара (на момент текущей даты).
  //Submission status
  /// </summary>
-    internal class Product
+    internal class Product : IComparable
```
Comment (VladimirVolkovD) [inline ConsoleApp1/HomeWork_lecture9_Collections/Product.cs:20]: Better here use IComparable<Product>, but IComparable also will work but with boxing

**Comment by MariaOdinets** [comment]
+ 1.3

File: ConsoleApp1/HomeWork_lecture9_Collections/Product.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            ProductName = productName;
+            ProductPrice = productPrice;
+            ManufactureDate = manufactureDate;
+            ExpirationDate = expirationDate;
+        }
+        public override string ToString()
+        {
+            return $"Item: {ProductName}, cost: {ProductPrice}, mfd: {ManufactureDate}, expiration: {ExpirationDate}";
+        }
```
Comment (VladimirVolkovD) [inline ConsoleApp1/HomeWork_lecture9_Collections/Product.cs:38]: add empty line

File: ConsoleApp1/HomeWork_lecture9_Collections/Product.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        public void CheckExpirationDate()
+        {
+            if (ExpirationDate.Date < DateTime.Now)
+            {
+                ExpirationStatus = "Product is spoiled";
+            }
+            else ExpirationStatus = "Product is fresh";
```
Comment (VladimirVolkovD) [inline ConsoleApp1/HomeWork_lecture9_Collections/Product.cs:45]:
```suggestion
            else
            {
                   ExpirationStatus = "Product is fresh";
            }
```

**Review by VladimirVolkovD — APPROVED** [review]

**Pull request merged**